### PR TITLE
open-iscsi: adding mutual authentication support and updating authentication parameters description

### PR DESCRIPTION
**Pull request by ricsanfre**
##### SUMMARY

Updating confusing description of authentication parameters included in documentation section
and adding the capability to specify mutual authentication between target and initiator

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/modules/system/open_iscsi.py

##### ADDITIONAL INFORMATION

node_auth, node_pass and node_user parameters are used for target login, not for target discovery. Authentication can be set for both procedures (discovery and login) but module current implementation does not support discovery authentication and thus parameters descriptions are confusing.

Adding two new module parameters (node_user_in, node_pass_in) to support mutual authentication.

### CODE REVIEW

File: plugins/modules/system/open_iscsi.py [hunk 1]
```diff
@@ -200,6 +210,15 @@ def target_login(module, target, portal=None, port=None):
...
             module.run_command(cmd, check_rc=True)
 
+    if node_user_in:
+        params = [('node.session.auth.username_in', node_user_in),
+                  ('node.session.auth.password_in', node_pass_in)]
+        for (name, value) in params:
+            cmd = '%s --mode node --targetname %s --op=update --name %s --value %s' % (iscsiadm_cmd, target, name, value)
+            (rc, out, err) = module.run_command(cmd)
+            if rc > 0:
+                module.fail_json(cmd=cmd, rc=rc, msg=err)
```
Comment (Ajpantuso) [inline plugins/modules/system/open_iscsi.py:220]:
```suggestion
            module.run_command(cmd, check_rc=True)
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) as well.

File: plugins/modules/system/open_iscsi.py [hunk 2]
```diff
@@ -41,16 +41,24 @@
...
         type: str
     node_pass:
         description:
-        - The value for C(discovery.sendtargets.auth.password).
+        - The value for C(node.session.auth.password).
+        type: str
+    node_user_in:
+        description:
+        - The value for C(node.session.auth.username_in).
+        type: str
```
[the remark this answers is not in the document]
Comment (ricsanfre) [reply plugins/modules/system/open_iscsi.py:58]: Updated

File: plugins/modules/system/open_iscsi.py [hunk 3]
```diff
@@ -41,16 +41,24 @@
...
+    node_pass_in:
+        description:
+        - The value for C(node.session.auth.password_in).
         type: str
```
[the remark this answers is not in the document]
Comment (ricsanfre) [reply plugins/modules/system/open_iscsi.py]: Updated

[the remark this answers is not in the document]
Comment (ricsanfre) [reply plugins/modules/system/open_iscsi.py]: changelog fragment also created

File: changelogs/fragments/3422-open-iscsi-mutual-authentication-support.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - open-iscsi - Updating confusing parameters descriptions node_user and node_pass (https://github.com/ansible-collections/community.general/pull/3422).
```
Comment (Ajpantuso) [inline changelogs/fragments/3422-open-iscsi-mutual-authentication-support.yaml:3]:
```suggestion
```
Changelog items are not required for doc only changes.

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Other than the superfluous Changelog item this looks good to me.

File: changelogs/fragments/3422-open-iscsi-mutual-authentication-support.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - open-iscsi - Adding support for mutual authentication between target and initiator (https://github.com/ansible-collections/community.general/pull/3422).
```
Comment (Ajpantuso) [inline changelogs/fragments/3422-open-iscsi-mutual-authentication-support.yaml:3]:
```suggestion
  - open-iscsi - adding support for mutual authentication between target and initiator (https://github.com/ansible-collections/community.general/pull/3422).
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
One last changelog item.

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@ricsanfre thanks for implementing this!
@Ajpantuso thanks for reviewing!

**Pull request merged**